rootio/streamers.go

```diff
@@ -166,6 +166,46 @@ func (tse *tstreamerElement) TypeName() string {
 	return tse.ename
 }
 
+func (tse *tstreamerElement) MarshalROOT(w *WBuffer) (int, error) {
+	if w.err != nil {
+		return 0, w.err
+	}
+	pos := w.Pos()
+
+	w.WriteVersion(tse.rvers)
+	tse.named.MarshalROOT(w)
+
+	w.WriteI32(tse.etype)
+	w.WriteI32(tse.esize)
+	w.WriteI32(tse.arrlen)
+	w.WriteI32(tse.arrdim)
+
+	if tse.rvers == 1 {
+		w.WriteStaticArrayI32(tse.maxidx[:])
+	} else {
+		w.WriteFastArrayI32(tse.maxidx[:])
+	}
+	w.WriteString(tse.ename)
+
+	if tse.etype == 18 {
```

Comment:
actually, this should be put before writing out `tse.etype`.
and we probably don't want to change `tse.etype` variable, just modify the value being written out. and only if the `tse.ename` is the "correct" one.

ie:
```go
etype := tse.etype
if etype == kBool && (tse.ename == "Bool_t" || tse.ename == "bool") {
    etype = kUChar
}
w.WriteI32(etype)
```

Reply:
alright ! 
Then the switch i did isn't totally correct, is it ?

Reply:
the `switch` is fine.